### PR TITLE
fix Visual Studio form designer

### LINKED ISSUE #4937: form designer in Visual Studio is broken
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
form designer cannot be used because of two recent changes:
1. GitModuleControl has been made abstract in 68e98e4a95c016e6e4ea0c5f550a37875dd0164c
image
2. there was a change in a2e1251477d8faa04b5ec1c27a2ca606a3ea8013 which added a specific workaround to `SearchForUICommandsSource`: this caused an infinite loop when the designer is opened in Visual Studio

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
- open GitExtensions.sln
- open RevisionGpgInfo with View Designer (Shift+F7)

**What is the expected behavior?**
form designer can be used

**Environment you encounter the issue:**

- GitExtensions version: master
- GIT version: irrelevant
- OS version: Windows 8.1
- Visual Studio 2017 Community 15.7

**Did this work in previous version of GitExtensions (which)?**
yes, before the two commits mentioned in the current behavior section

### PR DESCRIPTION
**Pull request by szaliszali**
Fixes #4937

Changes proposed in this pull request:
- make `GitModuleControl` non-abstract again
- fix infinite loop when opening form designer by reverting the commit which caused it (a2e1251477d8faa04b5ec1c27a2ca606a3ea8013)
- modify FormEdit to accept GitUICommands in its constructor (it requires the git module because it applies autocrlf setting to the copied text fragment)
 
Screenshots before and after (if PR changes UI):
- before 
image
- after 
image

What did I do to test the code and ensure quality:
- open form designer in VS
- open Repository->Git maintenance->Recover lost objects..., right click a row in the table, click View, select and copy a part of the text

Has been tested on (remove any that don't apply):
- Git 2.16.1
- Windows 8.1
- Visual Studio 15.7

### CODE REVIEW

File: GitUI/GitModuleControl.cs [hunk 1]
```diff
@@ -128,18 +128,7 @@ private void SearchForUICommandsSource()
                     }
                     else
                     {
-                        if (parent.Parent == null)
```
Comment (gerhardol) [inline GitUI/GitModuleControl.cs]: Did you find out why this was changed originally and why are this reverted?

**Comment by szaliszali** [comment]
@RussKie updated the description, the original commit was a2e1251477d8faa04b5ec1c27a2ca606a3ea8013

**Comment by RussKie** [comment]
@pmiossec your change introduced in a2e1251 seems to cause an infinite loop when opening form designer.

**Comment by szaliszali** [comment]
@RussKie fine, opened #4939 #4940 #4941

**Pull request closed without merging**